packages/synapse-react-client/src/components/TanStackTable/StyledTanStackTable.tsx

```diff
@@ -1,25 +1,59 @@
+import { flexRender, Row, Table } from '@tanstack/react-table'
 import { useMemo } from 'react'
-import { flexRender, Table } from '@tanstack/react-table'
 import {
   StyledTableContainer,
   StyledTableContainerProps,
 } from '../styled/StyledTableContainer'
-import { MemoizedTableBody, TableBody } from './TableBody'
+import { MemoizedTableBody, TableBody, TableBodyProps } from './TableBody'
 import {
   getColumnSizeCssVariable,
   getHeaderSizeCssVariable,
 } from './TanStackTableUtils'
+import { StyledTanStackTableSlotProps, StyledTanStackTableSlots } from './types'
 
-type StyledTanStackTableProps<T = unknown> = {
-  table: Table<T>
+export type StyledTanStackTableProps<TData = unknown, TRowType = Row<TData>> = {
+  table: Table<TData>
   styledTableContainerProps?: StyledTableContainerProps
   fullWidth?: boolean
-}
+  slots?: StyledTanStackTableSlots<TData, TRowType>
+  slotProps?: StyledTanStackTableSlotProps<TData, TRowType>
```

Comment:
In an effort to reduce duplicate code, I added `slots`/`slotsProps` props to `StyledTanStackTable`. I borrowed heavily from MUI's [slots design pattern](https://mui.com/x/common-concepts/custom-components/) since it seems to be a straightforward way to override/customize components wrapped by a complex component.

Reply:
This is my first exposure to the slots design pattern, and I can see the utility.  I think it was a good call to copy this pattern in this implementation